src/main/java/com/owncloud/android/lib/resources/files/ChunkedFileUploadRemoteOperation.java

```diff
@@ -315,4 +316,12 @@ private PutMethod createPutMethod(String uriPrefix) {
 
         return putMethod;
     }
+
+    private int CalcAssembleTimeout(File file) {
+        final double threeMinutes = 3.0 * 60 * 1000;
+        final int thirtySeconds = 30 * 1000;
+        final int thirtyMinutes = 30 * 60 * 1000;
+        int AssembleReadTimeout = Math.max(thirtySeconds , Math.min((int)(threeMinutes * file.length() / 1e9) , thirtyMinutes) );
```

Comment:
```suggestion
return Math.max(thirtySeconds , Math.min((int)(threeMinutes * file.length() / 1e9) , thirtyMinutes) );
```

Reply:
I'd also suggest extracting `file.length() / 1e9` into a new variable named `fileSizeInGb` to make it easier to read and make the algorithm clearer.